src/targets/gpu/include/migraphx/gpu/fp_conversion.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_FP_CONVERSION_HPP
+#define MIGRAPHX_GUARD_RTGLIB_FP_CONVERSION_HPP
+
+#include <migraphx/shape.hpp>
+#include <migraphx/op/fp_conversion.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+
+struct context;
+
+struct hip_fp_conversion
```

Comment:
If you change the device function to written as `void fp_conversion(hipStream_t stream, const argument& result, const argument& arg)`, then this can be written as:

```cpp
struct hip_fp_conversion : unary_device<hip_fp_conversion, device::fp_conversion>
{
};
```